src/ExtendedTableWidget.cpp

```diff
@@ -314,35 +322,42 @@ void ExtendedTableWidget::copy(const bool withHeaders)
     const QString fieldSepText = "\t";
     const QString rowSepText = "\r\n";
 
+    QString sqlInsertStatement = QString("INSERT INTO \"%1\" ( '").arg(m->currentTableName().toString());
     // Table headers
-    if (withHeaders) {
+    if (withHeaders || inSQL) {
         htmlResult.append("<tr><th>");
         int firstColumn = indices.front().column();
         for(int i = firstColumn; i <= indices.back().column(); i++) {
             QByteArray headerText = model()->headerData(i, Qt::Horizontal, Qt::DisplayRole).toByteArray();
             if (i != firstColumn) {
                 result.append(fieldSepText);
                 htmlResult.append("</th><th>");
+                sqlInsertStatement.append("', '");
             }
 
             result.append(escapeCopiedData(headerText));
             htmlResult.append(headerText);
+            sqlInsertStatement.append(escapeCopiedData(headerText));
         }
         result.append(rowSepText);
         htmlResult.append("</th></tr>");
+        sqlInsertStatement.append("') VALUES (\"");
     }
 
     // Table data rows
     for(const QModelIndex& index : indices) {
         // Separators. For first cell, only opening table row tags must be added for the HTML and nothing for the text version.
-        if (indices.first() == index)
+        if (indices.first() == index) {
             htmlResult.append("<tr><td>");
-        else if (index.row() != currentRow) {
+            sqlResult.append(sqlInsertStatement);
+        } else if (index.row() != currentRow) {
             result.append(rowSepText);
             htmlResult.append(rowSepHtml);
+            sqlResult.append("\");\r\n"+sqlInsertStatement);
```

Comment:
Not sure if ```\r\n``` is the correct thing here. Because it's not CSV or something like that it might be better to use ```\r\n``` only on Windows and ```\n``` for all other platforms.

Reply:
Mhmm do you have any mean do discriminate which line termination to use or should I write an #ifdef __WIN32 #else #endif block?

Reply:
In fact aren't C/C++ compilers converting `\n` to `\r\n` on Windows platforms? We are assumming that `\n` would produce a usual new line in several places. I would be surprised to know that those lines are not working under Windows.